diff --git a/node_modules/@dylankenneally/react-native-ssh-sftp/android/src/main/java/me/keeex/rnssh/RNSshClientModule.java b/node_modules/@dylankenneally/react-native-ssh-sftp/android/src/main/java/me/keeex/rnssh/RNSshClientModule.java
index 89a0053..97d813c 100644
--- a/node_modules/@dylankenneally/react-native-ssh-sftp/android/src/main/java/me/keeex/rnssh/RNSshClientModule.java
+++ b/node_modules/@dylankenneally/react-native-ssh-sftp/android/src/main/java/me/keeex/rnssh/RNSshClientModule.java
@@ -33,6 +33,7 @@ import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
@@ -300,14 +301,13 @@ public class RNSshClientModule extends ReactContextBaseJavaModule {
 
           callback.invoke();
 
-//        int charVal;
-          String line;
-          while (client._bufferedReader != null && (line = client._bufferedReader.readLine()) != null) {
+          byte[] buffer = new byte[4096];
+          int bytesRead;
+          while (client._bufferedReader != null && (bytesRead = in.read(buffer)) != -1) {
             WritableMap map = Arguments.createMap();
             map.putString("name", "Shell");
             map.putString("key", key);
-            map.putString("value", line + '\n');
-//          map.putString("value", String.valueOf(charVal));
+            map.putString("value", new String(buffer, 0, bytesRead, StandardCharsets.UTF_8));
             sendEvent(reactContext, "Shell", map);
           }
 
